test/e2e/ownerrefs_finalizers_test.go

```diff
@@ -104,39 +111,56 @@ var _ = Describe("Ensure OwnerReferences and Finalizers are resilient with Failu
 						framework.KubeadmControlPlaneOwnerReferenceAssertions,
 						framework.ExpOwnerReferenceAssertions,
 						VSphereKubernetesReferenceAssertions,
-						VSphereReferenceAssertions,
+						VSphereReferenceAssertions(),
 					)
 					// This check ensures that finalizers are resilient - i.e. correctly re-reconciled, when removed.
 					By("Checking that finalizers are resilient")
-					framework.ValidateFinalizersResilience(ctx, proxy, namespace, clusterName, clusterctlcluster.FilterClusterObjectsWithNameFilter(clusterName),
+					framework.ValidateFinalizersResilience(ctx, proxy, namespace, clusterName, FilterObjectsWithKindAndName(clusterName),
 						framework.CoreFinalizersAssertionWithLegacyClusters,
 						framework.KubeadmControlPlaneFinalizersAssertion,
 						framework.ExpFinalizersAssertion,
-						vSphereFinalizers,
+						vSphereFinalizers(),
 					)
 
-					// Stop periodic patch.
-					forceCancelFunc()
+					// Stop periodic patch if any.
+					if forceCancelFunc != nil {
+						forceCancelFunc()
+					}
 
 					// This check ensures that the resourceVersions are stable, i.e. it verifies there are no
 					// continuous reconciles when everything should be stable.
 					By("Checking that resourceVersions are stable")
-					framework.ValidateResourceVersionStable(ctx, proxy, namespace, clusterctlcluster.FilterClusterObjectsWithNameFilter(clusterName))
+					framework.ValidateResourceVersionStable(ctx, proxy, namespace, FilterObjectsWithKindAndName(clusterName))
 				},
 			}
 		})
 
 		// Delete objects created by the test which are not in the test namespace.
 		AfterEach(func() {
-			cleanupVSphereObjects(ctx, bootstrapClusterProxy)
+			if testMode == GovmomiTestMode {
+				cleanupVSphereObjects(ctx, bootstrapClusterProxy)
+			}
 		})
 	})
 })
 
+const (
+	// well know secret names from flavorgen / templates.
+	csiConfigSecretName     = "vsphere-config-secret" //nolint: gosec
+	cpiCredentialSecretName = "cloud-provider-vsphere-credentials"
+)
+
 var (
 	VSphereKubernetesReferenceAssertions = map[string]func(types.NamespacedName, []metav1.OwnerReference) error{
 		// Need custom Kubernetes assertions for secrets. Secrets in the CAPV tests can also be owned by the vSphereCluster.
-		"Secret": func(_ types.NamespacedName, owners []metav1.OwnerReference) error {
+		"Secret": func(s types.NamespacedName, owners []metav1.OwnerReference) error {
+			// When using vcsim CRS cannot be applied (not supported by the fake API server), so ignoring all the Secrets that should be deployed by CRS.
```

Comment:
Can we update the note to link to the PR that fixes this? (or potentially just directly drop it if the CAPI PR merges soon and this can be solved by a CAPI bump)

Reply:
https://github.com/kubernetes-sigs/cluster-api/pull/10756 should fix it, I'm ok to wait for the CAPI PR to merge first

Reply:
_approved_

Reply:
in CAPV E2E tests we are using CAPV published images, so we actually need to wait for 1.7.3 😅 
what about merging and then follow up, so we unblock the work on supervisor tests with runtime sdk?

Reply:
fixed usign nightly images